### DISCUSSION TITLE
2.3.10: Tasks to Perform After Moving /nsm To A Smaller Partition?

### DISCUSSION THREAD

**Discussion by GeorgiaM-honestly** [Security-Onion-Solutions/securityonion#2222] [Unsupported Versions] [answered]
Hello,

I'm using 2.3.10. I wanted to reduce the size of /nsm from 1.7TB to 900GB, and, finding that xfs does not support reduction in size, I created a new xfs filesystem of the size I desired. I then used a live environment to mount the old one read-only, the new one read-write, and used rsync -aPDHv --numeric-ids to copy everything from the old to the new.

After this I appropriately adjusted the fstab in the os/system partition. On boot, everything seems to work fine. However I have noticed two things: in graphana:

 - The PCAP retention has reset from almost 4 weeks to a few hours. I checked to see if /nsm/pcap is populated, and it very much is. I am also able to pull PCAP for dates prior to the move
 - The disk size total for /nsm seems to assume the old size

Are there any other tasks I should perform after these changes?

**Comment by weslambert** [comment]
Are you referring to the disk size total in Grafana?  If so, I believe that is set when you first install, so the data that graph pulls from has likely not been updated.

**Reply by GeorgiaM-honestly** [reply]
Yes, that is correct. I had tried to attach screenshots but I got an error (didn't have time to figure that out). I'm reading configuration info on graphana at:

https://grafana.com/docs/grafana/latest/administration/configuration/

However i'm not able to find where its config is here in 2.3.10. Any hints? I did find the json in graphana itself, and I can see what needs to be changed, I just do not know where to change it.

The pcap retention graph also reset, although there is still pcap going back roughly 4 weeks and it is usable. I'm less concerned about that one, but once I know where the config is, perhaps I can work that one out too!

Thanks for your time!

**Reply by bayden10** [reply]
Hi,
check here `/opt/so/conf/grafana/grafana_dashboards/standalone` and modify your required configuration. I had the same issue but after expanding my nsm. Modified max and absolute values to reflect the additional 100GB. (i.e. old value "max": "131114444000" new value "max": "231114444000")

      "title": "soict - Disk Used(/)",
      "type": "gauge"
    },
    {
      "cacheTimeout": null,
      "datasource": "InfluxDB",
      "fieldConfig": {
        "defaults": {
          "custom": {},
          "mappings": [
            {
              "id": 0,
              "op": "=",
              "text": "N/A",
              "type": 1,
              "value": "null"
            }
          ],
          "max": "231114444000",
          "min": 0,
          "nullValueMode": "connected",
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "rgba(50, 172, 45, 0.97)",
                "value": null
              },
              {
                "color": "rgba(237, 129, 40, 0.89)",
                "value": "204891555200.0"
              },
              {
                "color": "rgba(245, 54, 54, 0.9)",
                "value": "218002999600.0"
              }
``

Jim

**Reply by GeorgiaM-honestly** [reply]
That did it!! Thanks a lot!

For others in the future, I obtained the value by using df -B 1KB. That tells df to show the information in blocks of 1KB. Graphana expects it in bytes, so simply multiply by 1,000 (add 3 zeros) and that should do.

Refreshing the graphana panel in the web browser immediately showed the change.

**Reply by bayden10** [reply]
@weslambert 
> Are you referring to the disk size total in Grafana? If so, I believe that is set when you first install, so the data that graph pulls from has likely not been updated.

Hi,
Ok but how can this data be updated since it is only taking it from the initial install?

Jim

**Comment by GeorgiaM-honestly** [comment, marked as answer]
After reboot, it went back to the max value determined at install time. Someone else had the same issue:

https://github.com/Security-Onion-Solutions/securityonion/discussions/2246

**Reply by bayden10** [reply]
Hi,

yes that was me, I restarted so-grafana and it resorted back to the original install size. I also noticed that there is a process running about every 10min that updates that json file with the original config and also changes the permissions back to `socore:socore `where I had `chown root:root `in order for it to save in place.

Also tried `cp /opt/so/conf/grafana/grafana_dashboards/standalone/so-Standalone.json` to `/etc/grafana/grafana_dashboards/so-Standalone.json`  as stated in `/opt/so/conf/grafana/etc/dashboards/dashboard.yml` for configuration and it does not seem to pickup the json there.

Also tried to rerun `so-setup network `and noticed it is not seeing my newly extended LV nsm again. Something is not updating here in both cases.

Will keep digging into this.
Jim

**Reply by TOoSmOotH** [reply]
There is a pillar value on /opt/so/saltstack/local/pillar/data/standalonetab.sls which holds the original value. Change that to the new value and run so-grafana-restart and the new value will be permanent.

**Reply by GeorgiaM-honestly** [reply]
Oh, that did it! Thank you so much!!

**Reply by bayden10** [reply]
Thanks Mike.
Had also upgraded to 10CPUs from 6 so that was going to be my next question.
Will this dynamically adjust the PCAP retention ?

**Reply by GeorgiaM-honestly** [reply]
Bayden10,

When I reduced the size of my /nfs partition, in grafana the pcap retention indicator was reset as if the existing pcap had been wiped out. In reality the pcap was and is still there and I have been able to pull it. I have not discovered what to do about this one.

As for CPU count, I sometimes change my cpus (virtual machine) anywhere from 10 - 16 and the CPU usage graph which has a datapoint for how many CPUs there are is updated on its own.

**Reply by bayden10** [reply]
@GeorgiaM-honestly  Ok thanks, 
was not sure about the CPU adjustment as suspected it might be used in various other calculations if Grafana if reading that .sls file.
